NSS-Therapeutic-Raccoons/BangazonSite
# Tg 31 complete order

## Link to Ticket
Closes # [31](https://github.com/NSS-Therapeutic-Raccoons/BangazonSite/issues/31)

## Description of Proposed Changes
- Given the user is authenticated and the user is viewing their open order.
- When the user clicks the Complete Order button then the user should be presented with a view that allows them to select a payment type for the order.
-Given the user selects a payment option the user can click the Done button and then the payment type must be added to the order.
- Finally the user will be presented with a confirmation/thank you screen

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout TG-31-completeOrder
```

examine areas of the ticket and ensure that everything works properly
- Check that when you are logged in and click the Cart that you see your order and a Complete Order button
- Click the complete order button and see that you are presented with Payment Type options
- Check that if you try to click done without choosing a payment Type option that it prompts you to choose.
- When you have choosen a payment type and clicked done you should be redirected to a Thank you page.

## Impacted Areas in Application

List general components of the application that this PR will affect:

* OrdersController
* OrderPaymentTypeViewModel
* ThankYou View and ChoosePaymentType View
* Details view (added the complete order button)

## Mentions @username

@figamus
@ricky-bruner
@markhale94
@taylorlgulley
@askingalot

## Definition of Done

1. The project must be fully documented. This includes the following:
	* Comments for classes and methods added
	* Comments for models, controllers, and views added
2. The testing steps must completely work per ticket requirements.
3. Andy must smile approvingly and smirk with satisfaction following the smile.
